content/articles/statefull-vs-stateless-architectures/index.md

```diff
@@ -0,0 +1,116 @@
+
+
+### Advantages Stateless Architecture
+1.	A network's monitoring system does not need to read the information beyond a single request to comprehend it, which improves visibility. This is because, every request is handled in isolation.
+2.	They simplify the design of the server since most functionality is handled client-side.
+3.	It is very reliable as it recovers from partial failures easily. The server does not need to store any particular state or session; thus, a failed server can restart or transfer the client's request to another server of the same organization.
+4.	Scalability is achieved easily. Since the servers do not have to store session states, resources can be freed to handle other incoming requests.
+5.	They can easily handle multiple sessions simultaneously since every request is isolated.
+6.	Memory usage on the server-side is reduced. This is because the information used to handle requests is not stored there.
+
+### Disadvantages of Stateless Protocols 
+1.	Recurring poor network performances. This might occur when increasing the redundant data sent in sequences of requests. Remember, we mentioned that data is not left on the server in a stateless network architecture.
+ ### Advantages of Stateful Architecture
+1.	Stateful architecture may add a security advantage to systems. That is why many banks rely on it to allow users to make online transactions. Its disadvantages, however, outweigh its benefits.
+### Disadvantages of Stateful Architecture include:
+1.	It does not easily recover from partial failures. When a server fails, it cannot just restart or pass a user request to another server because it needs to hold the session state along with the details.
+2.	As much of the complex functionality is handled server-side, an intricate server design is needed to design the server.
+3.	Too much dependency between server and client that when one side fails, all connections are lost, and the running process has to start over.
+
+### Conclusion
+
+Majority of present systmes are designed using a stateless design. The statelsss design gained popularity from the server focused statefull way where clients were passive interfaces dependent on servers to perform majority of the computational needs. Developments in technology coupled with introduction of powerfull client computers allowed the shift to a stateless preferred environment that adds time to focus on the product development than on technical issues like server software implementation frequently encounetered in a statefull design. Stateless systems also enjoy the ability to scale in case a product or service explodes in a market.
```

Comment:
The sentences in this paragraph are very long. Shorten them for ease of readability.